tests/src/test/scala/ch/epfl/bluebrain/nexus/tests/kg/IdResolutionSpec.scala

```diff
@@ -0,0 +1,121 @@
+import akka.http.scaladsl.unmarshalling.PredefinedFromEntityUnmarshallers
+import ch.epfl.bluebrain.nexus.delta.kernel.utils.UrlUtils
+import ch.epfl.bluebrain.nexus.tests.BaseSpec
+import ch.epfl.bluebrain.nexus.tests.Identity.listings.{Alice, Bob}
+import ch.epfl.bluebrain.nexus.tests.iam.types.Permission.Organizations
+import io.circe.Json
+
+class IdResolutionSpec extends BaseSpec {
+
+  private val org1   = genId()
+  private val proj11 = genId()
+  private val proj12 = genId()
+  private val ref11  = s"$org1/$proj11"
+  private val ref12  = s"$org1/$proj12"
+
+  private def resource(id: String) =
+    json"""
+      {
+        "@context": {
+          "@vocab": "https://bluebrain.github.io/nexus/vocabulary/"
+        },
+        "@id": "$id",
+        "field": "value"
+      }
+        """
+  private val uniqueId             = "https://bbp.epfl.ch/neuron"
+  private val reusedId             = "https://bbp.epfl.ch/synapse"
+  private val resourcePayload1     = resource(uniqueId)
+  private val resourcePayload2     = resource(reusedId)
+
+  private val unauthorizedAccessErrorPayload =
+    jsonContentOf("iam/errors/unauthorized-access.json")
+
+  override def beforeAll(): Unit = {
+    super.beforeAll()
+
+    val setup = for {
+      _ <- aclDsl.addPermission("/", Bob, Organizations.Create)
+      _ <- adminDsl.createOrganization(org1, org1, Bob)
+      _ <- adminDsl.createProject(org1, proj11, kgDsl.projectJson(name = proj11), Bob)
+      _ <- adminDsl.createProject(org1, proj12, kgDsl.projectJson(name = proj12), Bob)
+    } yield ()
+
+    val createResources = for {
+      _ <- deltaClient.post[Json](s"/resources/$ref11/_/", resourcePayload1, Bob)(expectCreated)
+      _ <- deltaClient.post[Json](s"/resources/$ref11/_/", resourcePayload2, Bob)(expectCreated)
+      _ <- deltaClient.post[Json](s"/resources/$ref12/_/", resourcePayload2, Bob)(expectCreated)
+    } yield ()
+
+    (setup >> createResources).accepted
+  }
+
+  "Id resolution" should {
+    val iri = UrlUtils.encode("https://bluebrain.github.io/nexus/vocabulary/resource")
+
+    "lead to an authorization failure for a user without permission" in {
+      deltaClient.get[Json](s"/resolve/$iri", Alice) { (json, response) =>
+        response.status shouldEqual StatusCodes.Forbidden
+        json shouldEqual unauthorizedAccessErrorPayload
+      }
+    }
+
+    "lead to an authorization failure when trying to resolve a resource that does not exist" in {
+      deltaClient.get[Json](s"/resolve/unknownId", Bob) { (json, response) =>
+        response.status shouldEqual StatusCodes.Forbidden
+        json shouldEqual unauthorizedAccessErrorPayload
+      }
+    }
+
+    "resolve a single resource" in {
+      eventually {
+        deltaClient.get[Json](s"/resolve/${UrlUtils.encode(uniqueId)}", Bob) { (json, response) =>
+          response.status shouldEqual StatusCodes.OK
+          json shouldEqual resourcePayload1
```

Comment:
Not a very big deal but is there a way to make clearer that `uniqueId` is the id of `resourcePayload1`?

Reply:
Done 👍